### PR TITLE
Add SponsorUsers UseCase

### PR DESCRIPTION
**Pull request by adrianclay**
* Creates accounts for each user
  * Sends a confirmation email to the sponsor
  * Only allows government email addresses to sponsor new users

### CODE REVIEW

File: lib/sponsor_users.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        sponsee_address = Mail::Address.new(sponsee).address
+        sponsor_email(sponsor, sponsor_address, sponsee_address)
+      else
+        sponsee_phone_number = PhoneNumber.internationalise_phone_number(sponsee)
+        sponsor_phone_number(sponsor_address, sponsee_phone_number)
+      end
+    end
+  end
+
+  attr_reader :user_model, :valid_sponsees
```
Comment (chrisblackburn) [inline lib/sponsor_users.rb]: It look me a while to figure out what `valid_sponsees` was. Is it randomly hidden between two methods?

File: lib/sponsor_users.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    validate_sponsees(sponsees)
+    invite_sponsees(sponsor, sponsor_address)
+    send_confirmation_email(sponsor_address)
+  end
+
+private
+
+  def invite_sponsees(sponsor, sponsor_address)
+    valid_sponsees.each do |sponsee|
+      if is_valid_email(sponsee)
```
Comment (chrisblackburn) [inline lib/sponsor_users.rb]: is it more Ruby to use `valid_email?`?

Comment (adrianclay) [reply lib/sponsor_users.rb]: Agreed.

File: lib/sponsor_users.rb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        sponsee_phone_number = PhoneNumber.internationalise_phone_number(sponsee)
+        sponsor_phone_number(sponsor_address, sponsee_phone_number)
+      end
+    end
+  end
+
+  attr_reader :user_model, :valid_sponsees
+
+  def validate_sponsees(sponsees)
+    @valid_sponsees = sponsees.select do |sponsee|
```
Comment (chrisblackburn) [inline lib/sponsor_users.rb]: As this has an `attr_reader` I don't think you need the @ prefix. Also I think when it's accessed earlier it doesn't so should probably pick one style

File: lib/sponsor_users.rb [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      is_valid_email(sponsee) || is_valid_phone_number(sponsee)
```
Comment (chrisblackburn) [inline lib/sponsor_users.rb]: same question about the `valid_xx?` methods being Ruby naming convention

File: spec/lib/sponsor_users_spec.rb [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    context 'On production' do
+      let(:environment) { 'production' }
+      let(:plural_sponsor_confirmation_template_id) { '58e8ef4a-ca6b-40cd-81df-ec9c781fed56' }
+
+      it 'Sends a multiple user confirmation email to the sponsor' do
+        expect(a_plural_sponsor_confirmation_request).to have_been_made.once
+      end
+    end
+
+    context 'On staging' do
```
Comment (chrisblackburn) [inline spec/lib/sponsor_users_spec.rb]: Unclear of the purpose of this test. Is it double entry bookkeeping against the config values?

Comment (adrianclay) [reply spec/lib/sponsor_users_spec.rb]: Yep, it's asserting picking up the environment correctly.

**Comment by adrianclay** [comment]
I turned the `attr_reader` for `valid_sponsees` back into a parameter as it was causing confusionaroo.

**Pull request merged**